Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/CCLConfigurationRepository.kt

```diff
@@ -43,30 +44,30 @@ class CCLConfigurationRepository @Inject constructor(
      **/
     suspend fun updateCCLConfiguration(): UpdateResult = try {
         var updateResult = UpdateResult.NO_UPDATE
+
         internalData.updateBlocking {
             Timber.tag(TAG).d("Updating ccl configuration")
 
-            val rawData = cclConfigurationServer.getCCLConfiguration()
-            val newConfig = rawData?.tryParseCCLConfigurations()
+            val downloadedConfigListRaw = cclConfigurationServer.getCCLConfiguration() ?: run {
+                // no new config was downloaded
+                Timber.tag(TAG).d("Nothing to update. Keeping old ccl config list")
+                return@updateBlocking this
+            }
 
-            if (rawData != null && newConfig == null) {
-                // parsing failed
+            val downloadedConfigList = downloadedConfigListRaw.tryParseCCLConfigurations() ?: run {
                 updateResult = UpdateResult.FAIL
+                return@updateBlocking this
             }
 
-            when (newConfig != null && newConfig != this) {
-                true -> {
-                    Timber.tag(TAG).d("Saving new config data")
-                    cclConfigurationStorage.save(rawData = rawData)
-                    updateResult = UpdateResult.UPDATE
-                    newConfig
-                }
-
-                false -> {
-                    Timber.tag(TAG).d("Nothing to update. Keeping old ccl config list")
-                    this
-                }
-            }
+            Timber.tag(TAG).d("Saving new config data")
+            downloadedCclConfigurationStorage.save(rawData = downloadedConfigListRaw)
```

Comment:
We still only store the downloaded config, but we update the `HotDataFlow` with the merged config (see below)